packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -894,16 +894,31 @@ class FormSectionComponent extends React.Component<Props> {
             (field.type === FIELD_WITH_DYNAMIC_DEFINITIONS &&
               getFieldType(field as IDynamicFormField, values) === DATE)
 
-          if (
-            isDateField &&
-            touched[`${field.name}-dd`] !== undefined &&
-            touched[`${field.name}-mm`] !== undefined &&
-            touched[`${field.name}-yyyy`] !== undefined
-          ) {
-            touched[field.name] =
-              touched[`${field.name}-dd`] &&
-              touched[`${field.name}-mm`] &&
-              touched[`${field.name}-yyyy`]
+          const isDateRangePickerField = field.type === DATE_RANGE_PICKER
+
+          const dateFields = [
+            `${field.name}-dd`,
+            `${field.name}-mm`,
+            `${field.name}-yyyy`
+          ]
+          const dateRangeFields = [
+            `${field.name}exact-dd`,
+            `${field.name}exact-mm`,
+            `${field.name}exact-yyyy`
+          ]
+
+          const areFieldsTouched = (fields: any[]) =>
+            fields.every((field) => touched[field] !== undefined)
+
+          const areFieldsValid = (fields: any[]) =>
+            fields.every((field) => touched[field])
```

Comment:
```suggestion
          const areFieldsTouched = (fields: string[]) =>
            fields.every((field) => Boolean(touched[field]))
```
Do we need two separate functions?

Reply:
probably not